src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  constructor() {
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+
+  bite(herb) {
+    if (!herb.hidden && !(herb instanceof Carnivore)) {
```

Comment:
The condition `!(herb instanceof Carnivore)` is unnecessary because the task specifies that the `bite` method should only check if the object is an instance of `Herbivore` and is not hidden. Simplify the condition by removing this check.